moto/awslambda/responses.py

```diff
@@ -316,9 +316,27 @@ def _list_aliases(self, function_name: str) -> TYPE_RESPONSE:
         return 200, {}, json.dumps(result)
 
     def _create_function(self) -> TYPE_RESPONSE:
-        fn = self.backend.create_function(self.json_body)
-        config = fn.get_configuration(on_create=True)
-        return 201, {}, json.dumps(config)
+        function_name = self.json_body["FunctionName"].rsplit(":", 1)[-1]
+        try:
+            self.backend.get_function(function_name, None)
```

Comment:
You can create a dedicated exception in `exceptions.py`. As long as it extends `JsonRestError`, Moto should automatically take care of the formatting.
That means that this can be simplified to

```suggestion
            self.backend.get_function(function_name, None)
            raise FunctionAlreadyExists  # or whatever name you choose
```

Reply:
Thanks! This is much cleaner. I have added a commit to define the exception this way.

Reply:
FWIW I am not raising immediately as you suggested in order to avoid having that being trapped by the try block which is catching the expected "there's nothing here" error. Arguably it would be better to trap that condition explicitly and then raise FunctionAlreadyExists there, but I'm trying to weasel my way out of doing the research to determine the correct exception raised in this case...